### ISSUE TITLE
Issues getting started on Scarf (Rutherford Appleton Lab)

### ISSUE THREAD

**Issue by Linusfeder** [ComputationalRadiationPhysics/picongpu#4799]
Hi.
I am working on using Picongpu on the A100 gpu nodes on Scarf at RAL, and I'm having an error when I try to run the LWFA example. The "empty" example runs successfully. My stderr is below; let me know if I can provide anymore information.

cpu-bind=MASK - gn0003, task  0  0 [14370]: mask 0x3f0000003f set

The following have been reloaded with a version change:
  1) OpenSSL/.1.1 => OpenSSL/1.1

cpu-bind=MASK - gn0003, task  0  0 [16550]: mask 0x3f0000003f set
cpu-bind=MASK - gn0003, task  0  0 [16699]: mask 0x3f0000003f set
[cupla] Error: </home/vol02/scarf1102/src/picongpu/include/picongpu/../pmacc/particles/memory/buffers/MallocMCBuffer.tpp>:74 
/home/vol02/scarf1102/src/picongpu/thirdParty/cupla/alpaka/include/alpaka/mem/buf/BufUniformCudaHipRt.hpp(237) 'TApi::free(ptr)' A previous API call (not this one) set the error  : 'cudaErrorInvalidValue': 'invalid argument'!
Unhandled exception of type 'St13runtime_error' with message '[cupla] Error: not defined cuplaError', terminating
/home/vol02/scarf1102/src/picongpu/thirdParty/cupla/alpaka/include/alpaka/mem/buf/BufUniformCudaHipRt.hpp(237) 'TApi::free(ptr)' A previous API call (not this one) set the error  : 'cudaErrorHostMemoryNotRegistered': 'pointer does not correspond to a registered memory region'!
[gn0003:16699:0:16699] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
srun: error: gn0003: task 0: Segmentation fault (core dumped)
srun: launch/slurm: _step_signal: Terminating StepId=2158992.1

-Linus

**Comment by steindev** [comment]
Hey @Linusfeder, sorry for the late reply. Your issue actually looks a bit like an out-of-memory error, but this should not happen on A100 with the standard Laser Wakefield example. Therefore, I would guess that probably something is misconfigured in the way you acquire resources. But without access to the system, and without knowledge of the system's hardware, this might be tough to debug.

But first of all, are you running on latest `dev`?
And have you acquired 2 GPUs to run the laser wakefield example? Due to moving window this examples requires two MPI ranks (=2 GPUs in standard configuration OR setting the command line parameter `-r 2`)

How did you launch the PIConGPU simulation?

**Comment by Linusfeder** [comment]
Hi,
Thanks for the response.

I launched using this command:
tbg -s sbatch -c etc/picongpu/1.cfg -t etc/picongpu/scarf/A100.tpl $HOME/runs/myLWFA03

I thought I had successfully bound to two GPUs, but maybe not. Is that not what the line cpu-bind means?
Is there a simple way to set PiConGPU output to say how many GPUs/CPUs it is running on?

[stderr.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/14195645/stderr.txt)
[A100.tpl.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/14195646/A100.tpl.txt)

I've attached my stderr file and .tpl file. I'm sure I've done something silly with the submission. I added .txt to upload them.

**Comment by BrianMarre** [comment]
number of gpus is usually set in your input config file, in your case in <path to input directory>/etc/picongpu/1.cfg. Our standard 1.cfg for the LaserWakefield exampleuse only 1 gpu, see [here](https://github.com/ComputationalRadiationPhysics/picongpu/blob/def5e82f8cda083b14948477de079bf11fdaa5ef/share/picongpu/examples/LaserWakefield/etc/picongpu/1.cfg#L35-L37)

**Comment by steindev** [comment]
@Linusfeder I sent you an email with a proposal how to proceed. Here are some things to try in the meantime:
As far as I can see from the [cluster description](https://www.scarf.rl.ac.uk/scarf_hardware.html), your `tpl` needs to be changed in a number of positions:
line 58: `.TBG_gpusPerNode=`if [ $TBG_tasks -gt 4 ] ; then echo 4; else echo $TBG_tasks; fi``
line 62: `.TBG_coresPerGPU=8`
But this should not have hurt you. So the issue must be somewhere else.
Except for the fact, that you where not asking the batch system for any memory. Then again, Scarf says they apply a default.

The idea would be to, first, acquire an interactive job on the cluster.
`srun  --time=00:30:00 --ntasks=1 --gres=gpu:a100:1 --cpus-per-task=8  --mem-per-cpu=8G -p gpu --pty bash`
Then compile the KHI (KelvinHelmholtz Instability) example and start it from the command line (without `tbg`) and see whether the error still appears.
To start it, do in your setup directory (after compiling) `./bin/picongpu -d 1 1 1 -g 96 256 12 -s 500 --periodic 1 1 1 --e_macroParticlesCount.period 100 --fields_energy.period 250 2>err.txt | tee out.txt`
Check the created file `err.txt` for error messages.
Then, please report back.

**Comment by Linusfeder** [comment]
Hi,
Thanks for your help. The KHI example ran without errors. I've also fixed the errors you pointed out in my .tpl file. I'm happy to try anything else you suggest or just wait for a VC session.
Thank you!

**Comment by BrianMarre** [comment]
If the KHI example ran without an error, did you try the LaserWakefield example again with the new `.tpl`?

**Comment by psychocoderHPC** [comment]
There is a high possibility that the `tpl` file is not correct and you spawn more than 1 MPI rank.
Pleas remove for testing in the tpl `srun -K1 !TBG_dstPath/input/bin/picongpu !TBG_author !TBG_programParams` by
`srun -K1 hostname`

If you start the job the host name should be only once in the file `stdout`. If it is visible more than once you have configured the slurm wrongly.
Please also send us the file `tbg/submit.start` (generarted by `tbg`). This is the file whcih is given to the batch system.

**Comment by PrometheusPi** [comment]
@Linusfeder Could you solve your problem in the meantime?

**Comment by steindev** [comment]
Private Communication w/ @Linusfeder: Can close for now. Will reopen or open new issue in case other issues appear.
